### ISSUE TITLE
Pagination missing from /blog/ archives

### ISSUE THREAD

**Issue by thegulshankumar** [WP2Static/static-html-output-plugin#59]
Normally, it detects properly all pages in the consecutive order.

```
/page/2/
/page/3/
/page/4/
/page/5/
/page/6/
...
```

However, many pro internet marketer prefer static homepage to capture emails, while having blog posts at `/blog/` that's why it's important to look into this problem

![image](https://user-images.githubusercontent.com/23408127/82633583-15147880-9c19-11ea-8b63-38e49d19e6f1.png)

Video shows some pages missing https://youtu.be/7HgZboXiGSo

Rest everything is awesome.

Testing Env
-----------------------
Lokl WordPress
Deploying to BunnyCDN
Active theme: Twenty Twenty

**Comment by leonstafford** [comment]
[postspaginationtestfix.zip](https://github.com/WP2Static/static-html-output-plugin/files/4677741/postspaginationtestfix.zip)


This one should be good for `posts` pagination URLs.

Need to ensure others are similarly OK:

 - [ ] tags & tag pagination
 - [ ] categories & tag pagination
 - [ ] pages and page pagination
 - [ ] custom post types and paginations
 - others??

**Comment by leonstafford** [comment]
[jclondon.zip](https://github.com/WP2Static/static-html-output-plugin/files/4677926/jclondon.zip)

quick build for jclondon user

**Comment by thegulshankumar** [comment]
Hi @leonstafford 

I have tested postspaginationtestfix.zip [as mentioned here](https://github.com/WP2Static/static-html-output-plugin/issues/59#issuecomment-633608471)


* All test was done after using **Delete Deploy Cache** button.

* These numeric pages from 1 to 16 is not required. It shows same index.html in every directories.
![image](https://user-images.githubusercontent.com/23408127/82865744-1e695200-9f45-11ea-873e-ecafb88ba79b.png)


* When I go to /blog/page/ there it was suppose to have `/blog/page/index.html` then `/blog/page/2/`, then `/blog/page/3/`

* but after page 2 to 16 is missing.

![image](https://user-images.githubusercontent.com/23408127/82866047-c252fd80-9f45-11ea-93d6-55182a193785.png)

* Here's the list of Detected URLS
https://pastebin.com/raw/iM8ugkum

* 404 logs https://pastebin.com/raw/Q3bqCpsY

Demo: `https://pagination.b-cdn.net/blog/`

I could carefully repro this issue multiple times with **ZIP**, Netlify and Bunny Deployment.

Thanks

**Comment by thegulshankumar** [comment]
Update: **Category** and **Tag** pagination are fine. The problem is limited to defined 'Blog Page' pagination.

![image](https://user-images.githubusercontent.com/23408127/82867147-f62f2280-9f47-11ea-965c-e46d8e1c95b0.png)

**Comment by leonstafford** [comment]
ok, thanks - I think I know what I did wrong - removed the `/page/` (or custom pagination link if ppl change it from posts pagination url building. I just looked at the detected urls and saw a bunch of `/blog/1....n` urls and thought that looked good, not checking in my test site. Will fix that. 

Also, interesting behaviour by WP that any trailing url segment with a number renders without 404, but adding an extra segment with numbers or letters will 404, ie blog.com/blah/1 vs blog.com/blah/1/2
